src/lib/base.jl

```diff
@@ -115,3 +115,11 @@ end
     end
     return pairs(t), pairs_namedtuple
 end
+
+@adjoint function Base.getfield(p::Pair, i::Int)
```

Comment:
How to write the adjoint method for LinearAlgebra.diagm of Pair{Int64,Array{Float64,1}}